### PR TITLE
refactor: simplify reading config via sourcemap

### PR DESCRIPTION
**Pull request by Cafe137**
Continuation of https://github.com/ethersphere/swarm-cli/pull/79#discussion_r623898323

Instead of having duplicated `if` blocks for each option that may be specified via the swarm-cli config file, we just have to maintain an array: `CONFIG_OPTIONS`.

### CODE REVIEW

**Comment by Cafe137** [comment]
Actually, we still don't have tests for this behaviour, so let me write a few first.

File: test/command/config.spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const configFilePath = join(configFolderPath, configFileName)
+
+  beforeAll(() => {
+    global.console.log = jest.fn(message => {
+      consoleMessages.push(message)
+    })
+    jest.spyOn(global.console, 'warn')
+    //set config environment variable
+    process.env.SWARM_CLI_CONFIG_FOLDER = configFolderPath
+    process.env.SWARM_CLI_CONFIG_FILE = configFileName
```
Comment (nugaon) [inline test/command/config.spec.ts:20]: does it not conflict with the other test runs?

File: src/command/root-command/command-config.ts [hunk 2]
```diff
@@ -4,8 +4,14 @@ import { join } from 'path'
 import { exit } from 'process'
 import { beeApiUrl, beeDebugApiUrl } from '../../config'
 import { Identity } from '../../service/identity/types'
+import { ConfigOption } from '../../utils/types/config-option'
 import { CommandLog } from './command-log'
 
+export const CONFIG_OPTIONS: ConfigOption[] = [
```
Comment (nugaon) [inline src/command/root-command/command-config.ts:10]: Please write jsdoc about its role and how to handle

**Review by nugaon — APPROVED** [review]
LGTM!

File: test/command/config.spec.ts (hunk 1 above)
Comment (Cafe137) [reply test/command/config.spec.ts:20]: Test suites are supposed to have isolated envs

**Pull request merged**